tools/topology/topology2/include/components/ctc.conf

```diff
@@ -0,0 +1,81 @@
+#
+# Common widget ctc
+#
+# A generic ctc widget. All attributes defined herein are namespaced
+# by alsatplg to "Object.Widget.ctc.N.attribute_name"
+#
+# Usage: this component can be used by instantiating it in the parent object. i.e.
+#
+# 	Object.Widget.ctc."N" {
+# 		index		1
+# 		type		"effect"
+#	}
+#
+# Where N is the unique instance number for the parent object.
+
+Class.Widget."ctc" {
```

Comment:
@lgirdwood @ranj063 is there a way to set a min/max channel in widgets? I don't see anything in widget common. Might be useful for validation when pipelines define channels outside the range their components support.

Reply:
Adding @singalsu too

Reply:
PCM has channels_min and channels_max but I don't know if it would here. In my experience when channels count can be other than default 2, it's better to list for the instances explicitly the in and out formats with possible channels counts (and all rates if non-48 kHz is possible). There's an arrays combiner than can help when there's many combinations.

Reply:
Makes sense but is there a way to define that at the widget level.

E.g. dev A makes comp 1
Dev B makes pipeline with comp 1
Dev B fails to notice restrictions on formats

If format restrictions were part of the widget definition then it would be easy to identify if all widget formats were a superset of the pipelines set of supported formats. If they were a subset then the build would fail. Just a thought on finding bugs before runtime.

Reply:
@cujomalainey audio formats are always part of individual widgets. A pipeline does not define a superset of audio formats at all. But its an interesting idea to identify discrepancies in audio formats across widgets within the same pipeline and fail during topology  build. Let me give it a thought.